cql3/statements/describe_statement.cc

```diff
@@ -620,6 +738,13 @@ future<std::vector<std::vector<bytes_opt>>> schema_describe_statement::describe(
 
     auto result = co_await std::visit(overloaded_functor{
         [&] (const schema_desc& config) -> future<std::vector<description>> {
+            if (config.with_salted_hashes) {
```

Comment:
any user with right permissions could call this (and describe with internals) ? have you considered allowing only superuser?

maybe you talked about this already somewhere, I don't recall it from design doc

Reply:
It wasn't mentioned in the doc, no, but that's a good point. I agree the user should have the write permission to be able to execute that. I'll change it

Reply:
my question is even broader, do we ever want non super user to use this? it's potentially dangerous api

Reply:
CC @piodul

Reply:
> It wasn't mentioned in the doc, no, but that's a good point. I agree the user should have the write permission to be able to execute that. I'll change it

I stand corrected. It *was* mentioned in the doc:

> Because of that, it is very important that this statement is only allowed to be executed if the user that issues the DESCRIBE is granted **read** permissions to the `system.roles` table.

It's still good to discuss whether we should change the required permission to being a superuser.